diff --git a/oslo_log/pipe_mutex.py b/oslo_log/pipe_mutex.py
index 08a1fe3..3aa4832 100644
--- a/oslo_log/pipe_mutex.py
+++ b/oslo_log/pipe_mutex.py
@@ -136,6 +136,13 @@ class PipeMutex:
         # do, so nobody does it and that's okay.
         self.close()
 
+    def __enter__(self):
+        self.acquire()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.release()
+
 
 def pipe_createLock(self):
     """Replacement for logging.Handler.createLock method."""
